.eslintrc.js

```diff
@@ -1,13 +1,20 @@
 module.exports = {
-  extends: ["prettier"],
+  parser: "@typescript-eslint/parser", // Specifies the ESLint parser
+  parserOptions: {
+    ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
+    sourceType: "module" // Allows for the use of imports
+  },
   env: {
     es2020: true,
-    node: true,
-  },
-  parser: "@typescript-eslint/parser",
-  parserOptions: {
-    ecmaVersion: 11,
-    sourceType: "module",
+    node: true
   },
-  plugins: ["@typescript-eslint"],
+  extends: [
+    "plugin:@typescript-eslint/recommended", // Uses the recommended rules from the @typescript-eslint/eslint-plugin
+    "prettier/@typescript-eslint", // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
+    "plugin:prettier/recommended" // Enables eslint-plugin-prettier and eslint-config-prettier. This will display prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
```

Comment:
I am against this addition. Prettier shouldn't show up as an error. We already automatically fix it before commit and people can set up format on save.

Reply:
It pollutes the editor when writing.

Reply:
This change prefers ESLint with prettier plugin so it should be better for use with VS Code as it's the recommended way according to the VS Code team. What do you suggest ?

Reply:
It's super annoying to see ESLine errors and warnings that you know are going to be automatically formatted away when you hit save. I vote against it .

Reply:
Ah, now I get what you mean now. My only reason for this PR was to enforce default formatting rules but I don't care how :-)